Assets/Script/Gameplay/TextureManager.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using Cysharp.Threading.Tasks;
+using UnityEngine;
+using YARG.Helpers.Extensions;
+using YARG.Song;
+
+namespace YARG.Gameplay
+{
+    /// <summary>
+    /// A point of integration between GamePlay state
+    /// and venue materials
+    /// </summary>
+    public class TextureManager : GameplayBehaviour
+    {
+
+        // Smoothing factor (adjust to taste)
+        [Range(0.0f, 0.95f)]
+        public float fftSmoothingFactor = 0.8f;
+        [Range(0.0f, 0.95f)]
+        public float waveSmoothingFactor = 0.5f;
+
+        private Texture2D _sourceIcon;
+        private Texture2D _albumCover = null;
+        private Texture2D _soundTexture = null;
+        private float[] _fft = new float[FFT_SIZE * 2];
+        private float[] _wave = new float[FFT_SIZE * 2];
+        private float[] _prevFft = new float[FFT_SIZE * 2];
+        private float[] _prevWave = new float[FFT_SIZE * 2];
+
+        private static int _soundTexId = Shader.PropertyToID("_Yarg_SoundTex");
+        private static int _sourceIconId = Shader.PropertyToID("_Yarg_SourceIcon");
+        private static int _albumCoverId = Shader.PropertyToID("_Yarg_AlbumCover");
+
+        private const double MIN_DB = -100.0;
+        private const double MAX_DB = -30.0;
+        private const double DB_RANGE = MAX_DB - MIN_DB;
+        private const int FFT_SIZE_LOG = 9 /* aka log2(512) */;
```

Comment:
Firefox uses a default FFT size of 2048. If we want to ease the porting of ShaderToy shaders, this needs to change.

```suggestion
        private const int FFT_SIZE_LOG = 11 /* aka log2(2048) */;
```